.github/workflows/main.yml

```diff
@@ -54,3 +55,27 @@ jobs:
         with:
           name: python-artifacts
           path: .
+  docker-ci:
+    if: ${{ github.ref_name == 'main' }}
```

Comment:
The `docker-ci` job is set to run only if the branch is 'main'. Consider if this is the desired behavior, especially for testing purposes. You might want to allow this job to run on other branches as well.